Sources/EdgeNetworkHandlers/NetworkResponseHandler.swift

```diff
@@ -38,31 +36,47 @@ class NetworkResponseHandler {
     func addWaitingEvents(requestId: String, batchedEvents: [Event]) {
         guard !requestId.isEmpty, !batchedEvents.isEmpty else { return }
 
-        let eventIds = batchedEvents.map { $0.id.uuidString }
         serialQueue.sync {
             if self.sentEventsWaitingResponse[requestId] != nil {
                 Log.warning(label: self.LOG_TAG, "Name collision for requestId \(requestId), events list is overwritten.")
             }
 
-            self.sentEventsWaitingResponse[requestId] = eventIds
+            let uuids = batchedEvents.map { $0.id.uuidString }
+            let timestamps = batchedEvents.map { $0.timestamp }
+            self.sentEventsWaitingResponse[requestId] = zip(uuids, timestamps).map { ($0, $1) }
         }
     }
 
+    /// Adds the requestId in the internal `sentEventsWaitingResponse` with the associated event.
+    /// If the same requestId was stored before, the new list will replace the existing event(s).
+    /// - Parameters:
+    ///   - requestId: batch request id
+    ///   - event: the event sent to ExEdge
+    func addWaitingEvent(requestId: String, event: Event) {
+        addWaitingEvents(requestId: requestId, batchedEvents: [event])
+    }
+
     /// Remove the requestId in the internal `sentEventsWaitingResponse` along with the associated list of events.
     /// - Parameter requestId: batch request id
     /// - Returns: the list of unique event ids associated with the requestId that were removed
     func removeWaitingEvents(requestId: String) -> [String]? {
         guard !requestId.isEmpty else { return nil }
 
-        return sentEventsWaitingResponse.removeValue(forKey: requestId)
+        return sentEventsWaitingResponse.removeValue(forKey: requestId)?.map({$0.uuid})
     }
 
     /// Returns the list of unique event ids associated with the provided requestId or empty if not found.
     /// - Parameter requestId: batch request id
     /// - Returns: the list of unique event ids associated with the requestId or nil if none found
     func getWaitingEvents(requestId: String) -> [String]? {
         guard !requestId.isEmpty else { return nil }
-        return sentEventsWaitingResponse[requestId]
+        return sentEventsWaitingResponse[requestId]?.map({$0.uuid})
+    }
+
+    /// Sets the last reset date
+    /// - Parameter date: a `Date`
+    func setLastReset(date: Date) {
+        lastResetDate.mutate {$0 = date}
```

Comment:
We may need to persist this value for the resetIdentities events that are queued persistently and handled in the next app session - to be handled in https://github.com/adobe/aepsdk-edge-ios/pull/178